pageserver/src/tenant/delete.rs

```diff
@@ -60,7 +58,7 @@ fn remote_tenant_delete_mark_path(
         .context("Failed to strip workdir prefix")
         .and_then(RemotePath::new)
         .context("tenant path")?;
-    Ok(tenant_remote_path.join(Utf8Path::new("deleted")))
+    Ok(tenant_remote_path.join(Utf8Path::new("timelines/deleted")))
```

Comment:
Where do we do the migration for tenants that already have the old `deleted` mark? 
Is it safe to not do a migration? I haven't though that through, please explain.

How about calling the mark `deleted_v2` then to make it a bit clearer that there was a mark at a previous location?
Prehaps leave a comment about where the previous location was

Also, as with the local state changes: we should acknowledge this is a breaking change, we can't roll back the software because the old software won't know about the new location (and I assume we haven't thought through what happens if the old software encounters a `timelines/deleted` key).

Reply:
> Where do we do the migration for tenants that already have the old deleted mark?
> Is it safe to not do a migration? I haven't though that through, please explain.

The note in the PR comment is:
```
- The remote deletion marker is moved into `timelines/` and detected as part of listing timelines rather than as a separate GET request.  If any existing tenants have a marker in the old location (unlikely, only happens if something crashes mid-delete), then they will rely on the control plane retrying to complete their deletion.
```

> How about calling the mark deleted_v2 then to make it a bit clearer that there was a mark at a previous location?

Hmm, I think we disambiguating these via the different path is sufficient -- if we call it v2 it's liable to make the code just a bit less readable in the long term as people wonder what v1 was.

> Also, as with the local state changes: we should acknowledge this is a breaking change, we can't roll back the software because the old software won't know about the new location (and I assume we haven't thought through what happens if the old software encounters a timelines/deleted key).

I think it's worth doing a quick PR to let the old version understand the new deletion marker so that we aren't totally incompatible.  We'll still have the rollback issue that local state will be interpreted as authoritative due to the lack of an attach marker, but that will be limited to any tenants that were in a transient state at the time we rolled back, and these can be fixed with a detach/attach cycle.

Reply:
I've opened https://github.com/neondatabase/neon/pull/5632 to give us a realistic possibility of rolling back if we have issues with this.

Reply:
#5632 merged.